### DISCUSSION TITLE
Text field HTML editor maximum length

### DISCUSSION THREAD

**Discussion by massimoliani** [directus/directus#9014] [Archive]
I would like to know if is possible to set a maximum length for a text field.
Like 4096 byte, the maximum length for a telegram message. 
I try to cust the string later but I can break html tags pairs and the customer can't know if the message he wrote is too long.

In the interface parameters editor is possible to insert a section for customazied format and settings.

I try to look for info and docs, but can't find.

Thanks.

**Comment by rijkvanzanten** [comment]
You can't set a length for a TEXT field. Use a String instead (which maps to a VARCHAR, which has a max length) 👍🏻

**Reply by massimoliani** [reply]
But varchar can’t use html editor interface. 
I would like just to limit the editor interface not the api. 
Can I try to add a before hook on create or update and send back an exception description too long?

**Comment by joselcvarela** [comment]
Hello @massimoliani
Like you said you can use custom hooks since you want to use WYSIWYG.

Here is an example:

```js
const { FailedValidationException } = require('directus');
const { isEmail, isLatLong, isURL } = require('validator').default;

function before_save_my_collection(operation) {
  return async function ({ payload }) {
    const new_payload = { ...payload };

    if (payload.email && !isEmail(payload.email))
      throw new FailedValidationException({
        message: 'Email has not a valid format',
        path: ['email'],
        type: 'custom.pattern.base',
      });

    return new_payload;
  };
}

function before_save(operation) {
  return async function (input, { collection, database, payload }) {
    if (collection === 'my_collection')
      return before_save_my_collection(operation)({ payload });

    return payload;
  };
}

module.exports = function registerHook() {
  return {
    'items.create.before': before_save('create'),
    'items.update.before': before_save('update'),
  };
};
```

**Reply by massimoliani** [reply]
Thank you so much

**Reply by massimoliani** [reply]
One more question @joselcvarela :
Is there a way to find more detailed documentation on available exception validator or hooks, interfaces etc customization?
Website docs are great to start but sometime need some extra info, may I look inside docs folder of development npm package?

**Reply by joselcvarela** [reply]
We are currently refactoring the documentation, so right now my suggestion is to give a look on code base.
Once the refactor is merged, we could improve it.

**Reply by massimoliani** [reply]
Thanks

**Reply by massimoliani** [reply]
Hi @joselcvarela I implement your suggestion and everything is working.

An issue for future releases is to change the error popup to have a pretty error display for at least some exception. To a nicer error display for end user.

I'm using the admin app as the app for end user using role and permissions. 

The error message is still to much for IT professionals. 😉
